db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,1922 @@
+              "2024-01-01T02:00:00Z",
+              ca,
+              mt
+          );
+          ExternalEvent aE = new ExternalEvent(
+              a.key() + "_event",
+              st,
+              a.key(),
+              dg,
+              "2024-01-01T1:10:00Z",
+              duration,
+              mt
+          ); // have to manually pick this
+          ExternalSource b = new ExternalSource(
+              "B",
+              st,
+              dg,
+              "2024-01-02T00:00:00Z",
+              "2024-01-01T00:00:00Z",
+              "2024-01-01T01:00:00Z",
+              ca,
+              mt
+          );
+          ExternalEvent bE = new ExternalEvent(b.key() + "_event", st, b.key(), dg, b.start_time(), duration, mt);
+
+          // verify the ranges are as expected
+          assertDoesNotThrow(() -> {
+            // insert generic external event type, source type, and derivation group
+            insertStandardTypes();
+
+            // insert sources
+            insertExternalSource(a);
+            insertExternalSource(b);
+
+            // insert events
+            insertExternalEvent(aE);
+            insertExternalEvent(bE);
+
+            var res = statement.executeQuery(
+                // language=sql
+                """
+                SELECT * FROM merlin.derived_events ORDER BY source_key
+                """
+            );
+
+            // verify the range for A is shorter than what is specified in the definition of "a" - it should start later.
+            final String[] expectedResults = {
+                "{[\"2024-01-01 01:00:00+00\",\"2024-01-01 02:00:00+00\")}",
+                "{[\"2024-01-01 00:00:00+00\",\"2024-01-01 01:00:00+00\")}"
+            };
+            compareLists(expectedResults, res, "source_range");
+          });
+        }
+      }
+
+      /**
+       * This test is an overlapping case wherein the source of higher precedence (the more recently valid one - B)
+       *    should shorten the range that A applies over. In this case, a source is succeeded after in time by a more
+       *    valid source.
+       *
+       *    A:  +++++++
+       *    B:     ++++++++
+       *        AAABBBBBBBB
+       */
+      @Test
+      void testBackwardOverlap() throws SQLException {
+        try(final var statement = connection.createStatement()) {
+
+          // create our sources and their per-window events
+          ExternalSource a = new ExternalSource(
+              "A",
+              st,
+              dg,
+              "2024-01-01T00:00:00Z",
+              "2024-01-01T00:00:00Z",
+              "2024-01-01T01:00:00Z",
+              ca,
+              mt
+          );
+          ExternalEvent aE = new ExternalEvent(
+              a.key() + "_event",
+              st,
+              a.key(),
+              dg,
+              "2024-01-01T00:00:00Z",
+              duration,
+              mt
+          ); // have to manually pick this
```

Comment:
What has to be manually picked? Comment is unclear, please refine

Reply:
Not entirely clear what was being referred to; presumably referencing that we have to explicitly encode when the events are. The comment has been removed.